central/image/service/http_handler.go

```diff
@@ -0,0 +1,153 @@
+package service
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/pkg/errors"
+	clusterUtil "github.com/stackrox/rox/central/cluster/util"
+	"github.com/stackrox/rox/central/role/sachelper"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/pkg/httputil"
+	"github.com/stackrox/rox/pkg/images/enricher"
+	"github.com/stackrox/rox/pkg/images/integration"
+	"google.golang.org/grpc/codes"
+)
+
+const (
+	NumberBytes = 100 * 1024
+	timeout     = 10 * time.Minute
+)
+
+type sbomRequestBody struct {
+	Cluster   string `json:"cluster"`
+	ImageName string `json:"image_name"`
+	Force     bool   `json:"force"`
+}
+
+type sbomHttpHandler struct {
+	integration      integration.Set
+	enricher         enricher.ImageEnricher
+	clusterSACHelper sachelper.ClusterSacHelper
+}
+
+var _ http.Handler = (*sbomHttpHandler)(nil)
+
+// SBOMHandler returns a handler for get sbom http request
+func SBOMHandler(integration integration.Set, enricher enricher.ImageEnricher, clusterSACHelper sachelper.ClusterSacHelper) http.Handler {
+	return sbomHttpHandler{
+		integration:      integration,
+		enricher:         enricher,
+		clusterSACHelper: clusterSACHelper,
+	}
+
+}
+
+func (h sbomHttpHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+
+	if r.Method != http.MethodPost {
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		return
+	}
+	// verify scanner v4 is enabled
+	if !features.ScannerV4.Enabled() {
+		httputil.WriteGRPCStyleError(w, codes.NotFound, errors.New("Scanner v4 is not enabled. Enabled scanner v4 to get SBOMs"))
+		return
+	}
+	if !features.SBOMGeneration.Enabled() {
+		httputil.WriteGRPCStyleError(w, codes.NotFound, errors.New("SBOM feature is not enabled"))
+		return
+	}
+	var params sbomRequestBody
+	reqSizeLimit := NumberBytes
+	reqSizeBytes := os.Getenv("ROX_SBOM_API_REQ_SIZE")
+	if reqSizeBytes != "" {
+		reqSizeLimitInt, err := strconv.Atoi(reqSizeBytes)
+		if err == nil {
+			reqSizeLimit = reqSizeLimitInt
+		}
+	}
+	// timeout api after 10 minutes
+	ctx, cancel := context.WithTimeout(r.Context(), time.Duration(timeout))
+	defer cancel()
+	//for testing only
+	log.Infof("request size is %d", reqSizeLimit)
+	lr := io.LimitReader(r.Body, int64(reqSizeLimit))
+	err := json.NewDecoder(lr).Decode(&params)
+	if err != nil {
+		httputil.WriteGRPCStyleError(w, codes.InvalidArgument, errors.Wrap(err, "decoding json request body"))
+		return
+	}
+	bytes, err := h.getSbom(ctx, params)
+	if err != nil {
+		httputil.WriteGRPCStyleError(w, codes.Internal, errors.Wrap(err, "error generating SBOM"))
```

Comment:
nit: consider removing the word `error` from the error message.

```suggestion
		httputil.WriteGRPCStyleError(w, codes.Internal, errors.Wrap(err, "generating SBOM"))
```

Reply:
done